# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/scheduler/tentative/yield/yield-abort.any.js
# Commit: 00dc9f4af854
# Full Hash: 00dc9f4af854b9ad0096aabf57e8a9cb19fda858
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Fix a bug where AbortController doesn't abort scheduler.yield r=smaug
#   
#   Instead of passing a single `AbortSignal` argument to `CreateTask`,
#   we now separate it into a `AbortSignal` and a `TaskSignal`. They
#   could be the same when `TaskController` is used, and if its
# ==============================================================================

diff -r 6934fa78ea13 -r 00dc9f4af854 testing/web-platform/tests/scheduler/tentative/yield/yield-abort.any.js
--- a/testing/web-platform/tests/scheduler/tentative/yield/yield-abort.any.js	Mon Mar 31 19:49:03 2025 +0000
+++ b/testing/web-platform/tests/scheduler/tentative/yield/yield-abort.any.js	Mon Mar 31 19:49:03 2025 +0000
@@ -20,4 +20,15 @@
     const p = scheduler.yield();
     await promise_rejects_dom(t, 'AbortError', p);
   }, {signal});
-}, 'yield() aborted in a separate task');
+}, 'yield() aborted by TaskController in a separate task');
+
+promise_test(t => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  return scheduler.postTask(async () => {
+    scheduler.postTask(async () => {controller.abort();}, {priority: 'user-blocking'});
+    t.step(() => assert_false(signal.aborted));
+    const p = scheduler.yield();
+    await promise_rejects_dom(t, 'AbortError', p);
+  }, {signal});
+}, 'yield() aborted by AbortController in a separate task');
